legacy/firmware/ethereum.c

```diff
@@ -701,9 +708,12 @@ void ethereum_signing_init(const EthereumSignTx *msg, const HDNode *node) {
 }
 
 void ethereum_signing_init_eip1559(const EthereumSignTxEIP1559 *msg,
-                                   const HDNode *node) {
+                                   const HDNode *node,
+                                   const EthereumDefinitionsDecoded *defs) {
   struct signing_params params = {
-      .chain_id = msg->chain_id,
+      .chain_id = defs ? defs->network.chain_id : msg->chain_id,
+      .chain_suffix =
+          defs ? defs->network.shortcut : get_ethereum_suffix(msg->chain_id),
```

Comment:
see comments on the other `signing_init`

Reply:
9556b8f18d0dab6fc84e20f31bad8ba707737562